docs/en/04_Changelogs/4.2.0.md

```diff
@@ -201,3 +201,147 @@ SilverStripe\Core\Injector\Injector:
+
+Enable caching for all page content (through `Page_Controller`).
+
+```diff
+<?php
+
+-use SilverStripe\Control\HTTP;
++use SilverStripe\Control\Middleware\HTTPCacheControlMiddleware;
+use SilverStripe\CMS\Controllers\ContentController;
+
+class PageController extends ContentController
+{
+    public function init()
+    {
+-        HTTP::set_cache_age(60);
++        HTTPCacheControlMiddleware::inst()
++           ->enableCache()
++           ->setMaxAge(60); // 1 minute
+        
+        parent::init();
+    }
+}
+```
+
+Note: SilverStripe will still override this preference when a session is active,
+a [CSRF token](/developer_guides/forms/form_security) token is present,
+or draft content has been requested.
+
+##### Opt-out for a particular controller action
+
+If a controller output relies on session data, cookies,
+permission checks or other triggers for conditional output,
+you can disable caching either on a controller level
+(through `init()`) or for a particular action.
+
+```diff
+<?php
+
+-use SilverStripe\Control\HTTP;
++use SilverStripe\Control\Middleware\HTTPCacheControlMiddleware;
+use SilverStripe\CMS\Controllers\ContentController;
+
+class PageController extends ContentController
+{
+    public function myprivateaction($request)
+    {
+-        HTTP::set_cache_age(0);
++        HTTPCacheControl::inst()
++           ->disableCache();
+        
+        return $this->myPrivateResponse();
+    }
+}
+```
+
+Note: SilverStripe will still override this preference when a session is active,
+a [CSRF token](/developer_guides/forms/form_security) token is present,
+or draft content has been requested.
+
+##### Global opt-in, ignoring session (advanced)
+
+This can be helpful in situations where forms are embedded on the website.
+SilverStripe will still override this preference when draft content has been requested.
+CAUTION: This mode relies on a developer examining each execution path to ensure
+that no session data is used to vary output. 
+
+Use case: By default, forms include a [CSRF token](/developer_guides/forms/form_security)
+which starts a session with a value that's unique to the visitor, which makes the output uncacheable.
+But any subsequent requests by this visitor will also carry a session, leading to uncacheable output
+for this visitor. This is the case even if the output does not contain any forms,
+and does not vary for this particular visitor.
+
+```diff
+<?php
+
+-use SilverStripe\Control\HTTP;
++use SilverStripe\Control\Middleware\HTTPCacheControlMiddleware;
+use SilverStripe\CMS\Controllers\ContentController;
+
+class PageController extends ContentController
+{
+    public function init()
+    {
+-        HTTP::set_cache_age(60);
++        HTTPCacheControl::inst()
++           ->enableCache($force=true) // DANGER ZONE
++           ->setMaxAge(60); // 1 minute
+
+        parent::init();
+    }
+}
+```
+
+#### Detailed Cache-Control Changes:
+
+ * Added `Cache-Control: no-store` header to default responses,
```

Comment:
I think this is a big one that needs looking at, I think the default is and should remain no-cache

Reply:
OK, no-store seems excessive. It would be of most use in cases where the content was too sensitive to leave on a user's system, but that's overkill for our default.